### PR TITLE
Fall back to 'tac' when 'rev' is not available

### LINKED ISSUE #29: Support on Windows OS
Is git-toolbelt supported on Windows 10 machines?

### PR DESCRIPTION
**Pull request by ernstki**
- should allow toolbelt to run in Git Bash on Windows
- closes #29 (hopefully)

### CODE REVIEW

File: git-modified [hunk 1]
```diff
@@ -113,7 +113,9 @@ if [ -z "$commit" ]; then
     fi
 else
     TAB="	" # literal tab char
-    git log -1 --name-status --pretty=format:"" "$commit" | cut -f2- | rev | cut -d"$TAB" -f1 | rev | make_relative | while read f; do
+    rev='rev'
+    type rev >/dev/null 2>&1 || rev='tac'
```
Comment (nvie) [inline git-modified:117]: Same request as in the other PR: let's not assume `tac` exists if `rev` does not, instead let's just do a test.  In this case, you could make it a helper function:

```sh
reverse () {
  if type rev >/dev/null 2>&1; then
    rev
  elif type tac >/dev/null 2>&1; then
    tac
  else
    echo "Core utility 'rev' (nor 'tac') not found." >&2
    exit 2
  fi
}
```

And then call `reverse` in the pipe line, like

```sh
git log ... | reverse | cut -d"$TAB" -f1 | reverse | ...
```

Comment (ernstki) [reply git-modified:117]: I can totally do that. I just wasn't sure initially which side of the human-friendly-wrapper-functions-vs-lines-of-code question you would come down on, so I tried to keep the changes small.

Do you have strong opinions about line length? I would really like to break pipes over multiple lines at ~78 columns, but if it doesn't bother you, I won't let it bother me.

Comment (nvie) [reply git-modified:117]: > I can totally do that. I just wasn't sure initially which side of the human-friendly-wrapper-functions-vs-lines-of-code question you would come down on, so I tried to keep the changes small.

Makes sense, it's a good default! In this case, the little extra helper function makes things easier to understand, so I'm ok with it :)

> Do you have strong opinions about line length? I would really like to break pipes over multiple lines at ~78 columns, but if it doesn't bother you, I won't let it bother me.

That's fine in this case. This one was getting quite long in particular 👍

**Comment by ernstki** [comment]
I had a serious misunderstanding about `tac` vs. `rev`; `tac` reverses linewise and `rev` reverses character-wise, without changing the order of the lines top-to-bottom.

They are not drop-in replacements for one another like I mistakenly assumed they were without _carefully_ reading the man pages.

I'm going to close this for now and reopen (with the other fixes you suggest already implemented) once I'm sure I'm doing the right thing here.

**Pull request closed without merging**